bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -0,0 +1,1049 @@
+
+        @Override
+        public void onData(@Nullable Stream stream, @Nullable DataFrame frame, @Nullable Callback callback) {
+            Objects.requireNonNull(frame);
+            Objects.requireNonNull(callback);
+            synchronized (this) {
+                content.append(frame.getData());
+                if (frame.isEndStream() && !completable.isDone()) {
+                    completable.complete(content.contentAsString().trim());
+                    content.reset();
+                }
+            }
+            callback.succeeded();
+        }
+
+        @Override
+        public boolean onIdleTimeout(@Nullable Stream stream, @Nullable Throwable x) {
+            handleHttp2Error(Http2Error.IDLE);
+            return true;
+        }
+
+        @Override
+        public void onTimeout(@Nullable Stream stream, @Nullable Throwable x) {
+            handleHttp2Error(Http2Error.TIMEOUT);
+        }
+    }
+
+    /**
+     * Class to collect incoming ByteBuffer data from HTTP 2 Data frames.
+     */
+    private static class DataFrameCollector {
+        private byte[] buffer = new byte[512];
+        private int usedSize = 0;
+
+        public void append(ByteBuffer data) {
+            int dataCapacity = data.capacity();
+            int neededSize = usedSize + dataCapacity;
+            if (neededSize > buffer.length) {
+                int newSize = (dataCapacity < 4096) ? neededSize : Math.max(2 * buffer.length, neededSize);
+                buffer = Arrays.copyOf(buffer, newSize);
+            }
+            data.get(buffer, usedSize, dataCapacity);
+            usedSize += dataCapacity;
+        }
+
+        public String contentAsString() {
+            return new String(buffer, 0, usedSize, StandardCharsets.UTF_8);
+        }
+
+        public Reader contentStreamReader() {
+            return new InputStreamReader(new ByteArrayInputStream(buffer, 0, usedSize), StandardCharsets.UTF_8);
+        }
+
+        public void reset() {
+            usedSize = 0;
+        }
+    }
+
+    /**
+     * Adapter for listening to SSE event stream events.
+     *
+     * It receives the incoming text lines. Receipt of the first data line causes the CompletableFuture to complete. It
+     * then parses subsequent data according to the SSE specification. If the line starts with a 'data:' message, it
+     * adds the data to the list of strings. And if the line is empty (i.e. the last line of an event), it passes the
+     * full set of strings to the owner via a call-back method.
+     *
+     * The stream must be permanently connected, so it ignores onIdleTimeout() events.
+     *
+     * The parent class handles most fatal errors, but since the event stream is supposed to be permanently connected,
+     * the following events are also considered as fatal:
+     *
+     * <li>onClosed()</li>
+     * <li>onReset()</li>
+     */
+    private class EventStreamListenerAdapter extends BaseStreamListenerAdapter<Boolean> {
```

Comment:
Since this file is rather large (> 1000 lines) consider extracting this class to another file.